### ISSUE TITLE
QMplay2/Win11 no longer connects to internet?

### ISSUE THREAD

**Issue by RJVB** [zaps166/QMPlay2#725]
Just noticed that QMPlay2 on our Win11 machine no longer connects to the internet, not for updating, not for YT videos, nothing.
I uninstalled and reinstalled the application; no luck. I added an explicit application rule to the firewall for it, to no avail. Maybe not surprising as there is no such rule on the other MSWin machine where QMPlay2 works normally. The only difference between the 2 machines is that the "broken" one has a UEFI firmware with secure boot enabled and the other an old BIOS ... and thus runs Win10.

The Win11 machine is recent but I do think that I verified that QMPlay2 worked.

I have to assume this is just me overlooking something or MSWin having changed something.

This can't have anything to do with Qt's "bearer" functionality, by chance?

**Comment by zaps166** [comment]
I don't use Qt for networking in QMPlay2. All network connections comes from FFmpeg. For Qt6 + Win10-only builds I used schannel instead of OpenSSL - maybe that's why it doesn't work for you. Could you check if qt5 package works on Win11?

**Comment by RJVB** [comment]
>I don't use Qt for networking in QMPlay2. All network connections comes from FFmpeg.

Also the check for updated versions? That one doesn't work either.
I suppose that yt_dlp also uses Python's network stack...

>Could you check if qt5 package works on Win11?

Sure - I'll have to sleep on it first ;)

**Comment by RJVB** [comment]
On Tuesday May 21 2024 15:34:45 Błażej Szczygieł wrote:
> Could you check if qt5 package works on Win11?

Sadly that doesn't make a difference. I keep getting connection errors from both the update check and yt-dlp. 

Is there a way to increase the logging verbosity (if the code even logs anything of potential interest)? Or an MSWin diagnostics feature I can use to see if I'm running into some kind of protection?

FWIW, I had to force the OS to run the Qt5-based installer despite the fact that the system is set to allow installation from any source.

**Comment by zaps166** [comment]
> Sadly that doesn't make a difference.

That's good, because it's not issue of my recent changes! :smiley: 

---

> Is there a way to increase the logging verbosity

QMPlay2 has poor logs. You can try to launch it with possibility to read stdout/stderr (I noticed it works on GUI apps when you launch it from e.g. Git Bash on Windows). Or you can modify EXE and change parameter to open CMD in hexeditor.

> FWIW, I had to force the OS to run the Qt5-based installer despite the fact that the system is set to allow installation from any source.

Maybe because of unknown developer and unsigned installer. But it's the same as QMPlay2 on Qt6 installer. Maybe somewhere in Windows Defender online database? :astonished: 

---

Could you launch yt-dlp from QMPlay2 directory from command line and check if it's working? Path: `%APPDATA%\QMPlay2\yt-dlp.exe`

**Comment by RJVB** [comment]
>. Or you can modify EXE and change parameter to open CMD in hexeditor.

What do you mean with that?

>Maybe because of unknown developer and unsigned installer. But it's the same as QMPlay2 on Qt6 installer. Maybe somewhere in Windows Defender online database? 

Hmmm. I didn't have the issue with the Qt6 installer, indeed.

>Could you launch yt-dlp from QMPlay2 directory from command line and check if it's working? Path: `%APPDATA%\QMPlay2\yt-dlp.exe`

I have QMPlay2 installed under \Program Files, could that make a difference?

Anyway, the yt-dlp  I have under %APPDATA%\Roaming\QMPlay2 is called yt-dlp_x86.exe, and managed to update itself just fine. And `yt-dlp ytsearch:bla` also found and downloaded a magnificent video ^^

**Comment by RJVB** [comment]
On Wednesday May 22 2024 07:54:51 Błażej Szczygieł wrote:
>That's good, because it's not issue of my recent changes! :smiley: 

OK, you can shoot me now ... I had clearly NOT tested QMPlay2's networking features on that PC ... The application settings had been transferred from a previous PC, but not the proxy server that I ran on that machine.

Mystery solved, sorry for the noise :-/

But since I learned there's now a Qt5 and a Qt6 version: is either preferable over the other, beyond the obvious latest,greatest argument?

**Comment by zaps166** [comment]
> OK, you can shoot me now ... I had clearly NOT tested QMPlay2's networking features on that PC ... The application settings had been transferred from a previous PC, but not the proxy server that I ran on that machine.

No problem, good that problem is solved xD So the proxy in QMPlay2 is working good, right?

> But since I learned there's now a Qt5 and a Qt6 version: is either preferable over the other, beyond the obvious latest,greatest argument?

Qt5 version is for Windows 7 and 8. I don't want to force Qt6 on Windows 7 and don't want to build 3 different packages, so I dropped Windows 7 support. IMHO better install Win10 (+ disable some telemetry, some background tasks and defender antivirus) on older machine instead of running Windows 7.

**Comment by RJVB** [comment]
On Wednesday May 22 2024 13:14:52 Błażej Szczygieł wrote:

>So the proxy in QMPlay2 is working good, right?

To the extent that you don't get any connectivity when there is no proxy at the configured address ;)
On the old PC I ran privoxy, which is obsolete and transparent if you don't want to set it up as a man in the middle.
>
>> But since I learned there's now a Qt5 and a Qt6 version: is either preferable over the other, beyond the obvious latest,greatest argument?

>. IMHO better install Win10 (+ disable some telemetry, some background tasks and defender antivirus) on older machine instead of running Windows 7.

Depends how much older! That old, dead PC (an i3 HP) had come with Win10 but the past few years you had to be REALLY patient after logging in or waking it from sleep. Upgrading RAM and moving to an SSD didn't change all that much. And let's now talk about my Acer Aspire One netbook at all. It runs Win10 because the original Win7 install (that worked pretty decently) got borked by a few unhappily placed bad HDD sectors. The upgrade must have taken 48 hours and every OS update long enough that I call it "overnight". Curiously it's really slower in getting ready after a login than the HP was.

But you didn't really answer my question ... does being for Win7+8 make it worse on Win11? I can't say I noticed any difference after "downgrading".

**Comment by blzos** [comment]
> Qt5 version is for Windows 7 and 8. I don't want to force Qt6 on Windows 7 and don't want to build 3 different packages, so I dropped Windows 7 support. IMHO better install Win10 (+ disable some telemetry, some background tasks and defender antivirus) on older machine instead of running Windows 7.

> This is probably the last release for Windows 7 and 8 and 8.1 and 10 32-bit!

Maybe (if this is still possible) leave the version for older machines on Windows 7, reducing the releases to the QMPlay2-Win64-24.04.07-qt5-portable.7z edition

Removing 32-bit and additional options by type: nojs, offline....

>https://github.com/zaps166/QMPlay2/issues/725#issuecomment-2125721113
> And let's now talk about my Acer Aspire One...

Windows 7 has one undeniable advantage: completely disabling the DWM (Desktop Windows Manager). Disabling the process unloads the integrated video card to a minimum (No Aero theme, Using the theme Basic). Starting with Windows 8 DWM is always on.


_For example, after two years, SimpleWall returned support for Windows 7. Since, first of all, the popularity of such products is ensured by the wide support of devices._ https://github.com/henrypp/simplewall/releases

**Comment by RJVB** [comment]
>Windows 7 has one undeniable advantage: completely disabling the DWM (Desktop Windows Manager). Disabling the process unloads the integrated video card to a minimum (No Aero theme, Using the theme Basic). 

Yes, and that undeniably (IMHO) looks a lot better on (smaller) normal-resolution screens. IIRC you had to turn the candy UI off though - and I doubt Aero is responsible for the harddisk being sollicited continuously.

Foobar2000 is indeed one of the things I use that old netbook for, connected over HDMI to a nice AV amp. It's still as good at "serving" audio as it was when I just had it.

**Comment by zaps166** [comment]
> The upgrade must have taken 48 hours and every OS update long enough

I never do Windows upgrade. Once installed it must work, because windows updates mechanism is really horrible (here Linux and macOS are way better). Maybe better reinstall or update from USB stick instead.

>  does being for Win7+8 make it worse on Win11? I can't say I noticed any difference after "downgrading".

Because it's the same. Differences:
- Qt 6.7.0 (will be 6.7.1 in next release) with "passthrough" DPR scaling and Windows 11 style
- Windows Schannel instead of OpenSSL 1.1.x
- no DirectDraw output (pointless with DWM, overlay doesn't work anymore)

> completely disabling the DWM

You can't disable DWM, because it's a window manager (just like disable xfwm4 or kwin on Linux). You can disable DWM compositing in Aero basic or classic style. QMPlay2 can automatically suspend DWM compositing on Windows 7 when you use DirectDraw output (to use overlay).

> Maybe (if this is still possible) leave the version for older machines on Windows 7, reducing the releases to the QMPlay2-Win64-24.04.07-qt5-portable.7z edition

It doesn't matter if it's portable or installer. I need to keep Qt 5.15.x and OpenSSL 1.1.x (Schannel and OpenSSL 3.x doesn't work to me on Windows 7). How about 32-bit?

**Comment by RJVB** [comment]
On Wednesday May 22 2024 16:04:35 Błażej Szczygieł wrote:

>I never do Windows upgrade.

I hadn't planned on doing it either, but it was a last resort for getting a fully functional OS again without having to reinstall everything. There was a lot that I had installed, and most of it just worked again after the upgrade.

**Comment by zaps166** [comment]
> completely disabling the DWM

QMPlay2 on full screen bypasses DWM compositing (it's default setting, should work on most up-to-date drivers).

---

Do you think I should still build packages for Windows 7?

**Comment by blzos** [comment]
> You can't disable DWM, because it's a window manager (just like disable xfwm4 or kwin on Linux). You can disable DWM compositing in Aero basic or classic style. QMPlay2 can automatically suspend DWM compositing on Windows 7 when you use DirectDraw output (to use overlay).

Yes, that’s right, disabling window compositing reduces the consumption of processor and video card resources to 0%, since their combination occurs within the system kernel, and not in a separate process


> It doesn't matter if it's portable or installer. I need to keep Qt 5.15.x and OpenSSL 1.1.x (Schannel and OpenSSL 3.x doesn't work to me on Windows 7). How about 32-bit?

Considering that Windows 7 is mostly run by geeks, a portable program will be enough, I see no problem in going into the archive and copying the folder with the program to the desired location. For newer systems, leave the installation option.



> How about 32-bit?

32-bit are processors up to Pentium 4D, 2004 year, сan be excluded. But for computers **(especially netbooks)** starting from Intel Core Duo (2004) to Ivy Bridge (2012), using Windows 7 64-bit is possible, and for them you need to leave the version with QT5, which now works very stably.



> Do you think I should still build packages for Windows 7?

If this is possible, from a code point of view, and it has not completely switched to QT6, **it should be released**, reducing the number of releases to portable 64-bit. for uses in media centers and netbooks on Win7

**Comment by RJVB** [comment]
>Do you think I should still build packages for Windows 7?

If you can and it's not a big burden, why not?
I still haven't really grasped what the problem is with just building a single win64 package that will run on Win11 and older. As I said I didn't notice any difference between the "legacy" version and the win10+ version.
The fact OpenSSL 3 doesn't work for you is probably something that's worth checking out; having the same SSL library on every platforms seems like an advantage to me.

You mentioned the widget style: I'm using Fusion for that anyway (I'll admit the win11 style looks less bad than the others, but those sliders ...). And I'll admit I've been thinking of suggestion you to check out the QtCurve style that I've contributed to, to see if you can configure it to your own taste. ;)

**Comment by zaps166** [comment]
Ok, I reverted DDraw commit and changed my compilation scripts to build packages with Qt5 and OpenSSL 1.1, too (installer and 64-bit portable). I'll keep 32-bit, too, because it's not a problem. I don't know if it works with Windows Vista (nobody cares of it). But please don't request for Windows XP support :rofl: 

There should be possible to force Qt6 to run on Windows 7, but less work is to build with Qt5.

> QtCurve

I used it in QMPlay (first version, audio player only), it was over 10 years ago...

> The fact OpenSSL 3 doesn't work for you is probably something that's worth checking out

Maybe never updated Windows 7 or bugged OpenSSL 3 compilation. It was long time ago when I tried it. Same Schannel doesn't work on my Windows 7, so I'll keep OpenSSL 1.1 for Windows 7 (it's working for sure).

> Yes, that’s right, disabling window compositing reduces the consumption of processor 

It was good on VMs, too. Luckily Xorg can still run without any compositing or with XPresent extension which is really fast and low latency to allow tear-free with transparency (like [this](https://gitlab.com/zaps166/qdre/-/tree/master/qdre-compositor)).

**Comment by zaps166** [comment]
Removal of Win7 was planned here for months: #650, nobody complained.

**Comment by RJVB** [comment]
>I hope nobody will cry after removing of VDPAU...

Whazzat? :)

The remarks about Foobar2000 reminded me of a good reason to do 32bit builds: they (used to) work better with Wine. A moot argument for QMPlay2 of course, but I still haven't found a similarly well-organised pure audio player for Linux as FB2000 with the UI Columns UI ... (I have columns for genre, composer, artist, album and on-disk folder, and FB2000 actually breaks up tags at semicolons so a composite composer or artist property will show up under the individual tags).

**Comment by zaps166** [comment]
I don't remember how FB2000 looks like. But how about deadbeef, audacious, clementine? Or add more columns to QMPlay2's playlist?

**Comment by RJVB** [comment]
>I don't remember how FB2000 looks like.

Very much like an older iTunes version, in fact. A left sidebar with a number of "shortcuts", as many half-height columns as you want with the different selection categories, and underneath a panel spanning the remaining width with the result of your selection.

>But how about deadbeef, audacious, clementine?

Can't remember deadbeef, I do know that I spent a certain time tinkering Audacious but then lost interest, and IIRC Clementine is not as configurable as I want either. Idem for VLC.

Fb2000 currently crashes though under my current Wine version, so I may need to start looking for an alternative.

>Or add more columns to QMPlay2's playlist?

How? And will it index "artists=A;B;C" under artist=A, artist=B and artist=C?

**Comment by RJVB** [comment]
With Wine's marvelous UI theme :)
![image](https://github.com/zaps166/QMPlay2/assets/1756884/dab39e5b-5bda-41d1-8a38-632a82b0f601)

**Comment by blzos** [comment]
> Removal of Win7 was planned here for months: #650, nobody complained.

Awareness does not come immediately. Many people don't read the forum. 

> Or add more columns to QMPlay2's playlist?

Today the playlist looks complete, the filename - time. Information on the file in information panel (current play), which is abundant. This part of the player looks solid, considering the main purpose of the player - video files.

_There was a time when foobar (through skins) looked like a Christmas tree with a million data in the window, now everything tends towards simplification (suprematism) and no one would think of displaying 20 parameters for 40 files in a playlist! This will be information that is redundant for the brain._

![foobar2000](https://github.com/zaps166/QMPlay2/assets/166966429/3db4bfcc-26d5-4c0f-8eb2-af5330b75789)

**Comment by RJVB** [comment]
>_There was a time when foobar (through skins) looked like a Christmas tree with a million data in the window, now everything tends towards simplification (suprematism) and no one would think of displaying 20 parameters for 40 files in a playlist! This will be information that is redundant for the brain._

The default FB2000 UI hurts my eyes and makes my brain shut off (I got to see it recently). My UI as displayed above displays exactly the information I will want to have available at a glance - if and when I want it.

The observant reader will notice that the columns are for selection (filtering) purposes.

I found an old copy of Nightingale on my system, which provides very much the same interface in a nicer jacket, but sadly the application has been abandoned and doesn't even provide a way to select the output device.

**Comment by RJVB** [comment]
Since we've gone off-topic here anyway: does anyone here have a non-paying solution to get control over what BT codec is used on MSWin? Or if not, at least a tip how to set the latency like you can do with PulseAudio?

(I haven't even looked at the price of the alternative A2DP driver; I consider the fact they link the license to the mobo abusive esp. since they won't even consider updating it after a mobo repair.)
